### PR TITLE
ISPN-7161 RemoteGetFailureTest.testOneOwnerSuspectedNoFilter failures

### PR DESCRIPTION
**Pull request by danberindei**
https://issues.jboss.org/browse/ISPN-7161

* Add delay after view installation
* Change asserts to include the response in the failure messages

### CODE REVIEW

File: core/src/test/java/org/infinispan/distribution/RemoteGetFailureTest.java [hunk 1]
```diff
@@ -198,30 +200,34 @@ public void testOneOwnerSuspectedNoFilter(Method m) throws ExecutionException, I
...
-      CompletableFuture<Map<org.infinispan.remoting.transport.Address, Response>> future = cache(0).getAdvancedCache().getRpcManager().invokeRemotelyAsync(owners, clusteredGet, rpcOptions);
+      RpcManager rpcManager = cache(0).getAdvancedCache().getRpcManager();
+      CompletableFuture<Map<Address, Response>> future = rpcManager.invokeRemotelyAsync(owners, clusteredGet, rpcOptions);
 
       assertTrue(arrival.await(10, TimeUnit.SECONDS));
 
       installNewView(cache(0), cache(0), cache(1));
 
+      // RequestCorrelator processes the view asynchronously, so we need to wait a bit for node 2 to be suspected
+      Thread.sleep(100);
```
Comment (rvansa) [inline core/src/test/java/org/infinispan/distribution/RemoteGetFailureTest.java]: Can't we add some listener and countdown here?

File: core/src/test/java/org/infinispan/distribution/RemoteGetFailureTest.java [hunk 2]
```diff
@@ -198,30 +200,34 @@ public void testOneOwnerSuspectedNoFilter(Method m) throws ExecutionException, I
...
       release1.countDown();
-      Map<org.infinispan.remoting.transport.Address, Response> responses = future.get();
+      Map<Address, Response> responses = future.get();
       long requestCompleted = System.nanoTime();
-      assertTrue(responses.get(address1).isSuccessful());
-      assertNotNull(((SuccessfulResponse) responses.get(address1)).getResponseValue());
-      assertTrue(responses.get(address2) instanceof CacheNotFoundResponse);
-      assertTrue(TimeUnit.NANOSECONDS.toSeconds(requestCompleted - requestAllowed) < timeout);
+
+      assertEquals(0, TimeUnit.NANOSECONDS.toSeconds(requestCompleted - requestAllowed), timeout);
```
Comment (rvansa) [inline core/src/test/java/org/infinispan/distribution/RemoteGetFailureTest.java]: Is this assertEquals(double expected, double actual, double delta)? A bit confusing. Then, the comparison there is `Math.abs(expected - actual) <= delta` while we want strict comparison: `actual < delta` (`expected` is 0)

File: core/src/test/java/org/infinispan/distribution/RemoteGetFailureTest.java (hunk 1 above)
Comment (danberindei) [inline core/src/test/java/org/infinispan/distribution/RemoteGetFailureTest.java]: I thought about subclassing `CommandAwareRpcDispatcher` and `CustomRequestCorrelator` to make `receiveView` synchronous, but I decided that it wasn't worth the effort.

File: core/src/test/java/org/infinispan/distribution/RemoteGetFailureTest.java (hunk 2 above)
Comment (danberindei) [inline core/src/test/java/org/infinispan/distribution/RemoteGetFailureTest.java]: This seemed the simplest way to log the actual duration after a failure, but I agree it's a bit confusing.

**Comment by slaskawi** [comment]
Integrated thanks!

**Pull request merged**